### PR TITLE
Upgrade Scanamo to 0.4.0

### PR DESCRIPTION
**Pull request by philwills**
There's a lot more hygiene that could be done, but this seems to work and I'd rather not have to deal with conflicts against *all* the files.

### CODE REVIEW

**Comment by philwills** [comment]
Fixes #13

File: app/controllers/Amigo.scala [hunk 1]
```diff
@@ -168,6 +175,8 @@ class Amigo(
       .via(EventSource.flow)
     Ok.chunked(source).as("text/event-stream")
   }
+
+  def exec[T](ops: ScanamoOps[T]) = Scanamo.exec(dynamo.client)(ops)
```
Comment (cb372) [inline app/controllers/Amigo.scala]: Make this private? Also a type annotation would be nice, as I have no idea what it returns.

File: app/data/BakeLogs.scala [hunk 2]
```diff
@@ -19,17 +25,7 @@ object BakeLogs {
...
-    val queryRequest = new QueryRequest(tableName)
-      .withKeyConditionExpression("#bakeId = :bakeId")
-      .withExpressionAttributeNames(Map("#bakeId" -> "bakeId").asJava)
-      .withExpressionAttributeValues(Map(":bakeId" -> BakeId.dynamoFormat.write(bakeId)).asJava)
-      .withScanIndexForward(true) // return oldest logs first
-    val items: Iterable[Option[ValidatedNel[DynamoReadError, BakeLog]]] =
-      dynamo.client.query(queryRequest).getItems.asScala.map { item => Scanamo.from[BakeLog](new GetItemResult().withItem(item)) }
-    items.flatMap(_.flatMap(_.toOption))
+  def list(bakeId: BakeId): ScanamoOps[Iterable[BakeLog]] = {
+    table.query('bakeId -> bakeId).map(_.flatMap(_.toOption))
```
Comment (cb372) [inline app/data/BakeLogs.scala]: Much better!

So Scanamo queries with `scanIndexforward = true` by default?

File: app/data/Bakes.scala [hunk 3]
```diff
@@ -26,40 +32,27 @@ object Bakes {
...
-      r <- recipe.toIterable
-      dbModel <- dbModels.flatMap(_.flatMap(_.toOption))
+      r <- recipes.findById(recipeId)
+      dbModels <- table.query(('recipeId -> recipeId).descending)
     } yield {
-      Bake.db2domain(dbModel, r)
+      for {
+        recipe <- r.toIterable
+        dbModel <- dbModels.flatMap(_.toOption).take(limit)
+      } yield Bake.db2domain(dbModel, recipe)
```
Comment (cb372) [inline app/data/Bakes.scala]: A for inside a yield looks a bit weird. Could make the second for-comp a local def?

**Comment by cb372** [comment]
Thanks, I've been waiting for you to do this for me 😁 

It's a shame you had to change all the DAO objects (`Bakes`, etc.) into classes. I quite liked having them as objects. How about storing the Scanamo `Table` instances in `Dynamo.Tables` and letting the DAO objects look them up implicitly, like they were doing for the table names.

Apart from that, looks fine to me.

**Pull request closed without merging**